docs/developer-docs/smart-contracts/maintain/resource-limits.mdx

```diff
@@ -57,6 +57,17 @@ The limits depend on the message type as shown in the following table.
 | Update calls execution threads, per subnet                                           | 4           |
 | Update calls execution threads, per canister                                         | 1           |
 
+## Performance limits
+
+The block rate varies from 0.75 blocks per second to 1.5 blocks per second depending on the subnet and the number of nodes in the subnet.
+
+Up to 1000 messages can be included in a block.
```

Comment:
What we could add

1. The messages are selected round-robin per canister from a pool of up to 10_000 messages. 
2. Nodes do not admit messages to their pool when its full.
3. Messages are deleted from the pool after they have been received by execution or after they expired.

I am not sure how much it helps the developers though... What do you think?

What I'd definitely omit is the fact that for the NNS subnet the limit is 400 messages per block, since ordinary developers cannot deploy there

Reply:
That seems too low-level for most developers.

Reply:
I agree with Ulan, this might be too low-level for this page. Maybe it would fit in [this reference page](https://internetcomputer.org/docs/current/references/message-execution-properties) though?

Reply:
I agree with Ulan, I think this it too low-level for this page. Maybe it would fit in [this reference page](https://internetcomputer.org/docs/current/references/message-execution-properties) though?

Reply:
Thanks, Ulan and Jessie! Let's keep it out then :)

I wouldn't add it to the other page, since it has nothing to do with execution.